### PR TITLE
fix: close popup dialogs on esc key / click outside (#15024)

### PR DESCRIPTION
**Pull request by echianese**
Corrected same issue for PDP gallery popup dialog and "Import Products" on cart page.
https://github.com/SAP/spartacus/issues/15024

### CODE REVIEW

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 1]
```diff
@@ -7,28 +14,39 @@ import {
...
     block: true,
-    autofocus: 'button',
-    focusOnEscape: true,
+    autofocus: false,
+    focusOnEscape: false,
   };
 
+ @HostListener('click', ['$event'])
+  handleClick(event: UIEvent): void { 
+    console.log("target name =" + (event.target as any).tagName);
```
Comment (cawmichae) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: Should these console logs statements be there?

**Comment by BenjaminEmiliani** [comment]
Hey @echianese I would suggest to change the title of the PR to match our convention. The template is:
type: explanation of change (#issue number)

List of types https://sap.github.io/spartacus-docs/commit-guidelines/#commit-message-format

Example for this PR: 
fix: close popup dialogs on esc key in a few places (#15024)

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html [hunk 2]
```diff
@@ -1,4 +1,8 @@
-<div #dialog>
+<div #dialog
+[cxFocus]="focusConfig"
+(esc)="dismissModal('gone')"
+class="AddedToCartDialogComponent"
```
Comment (BenjaminEmiliani) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html:4]: is this class needed?

Comment (BenjaminEmiliani) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: 1) I think the string input to dismissModal() can be improved to be a bit more descriptive.

2) Is the class `AddedToCartDialogComponent` needed?

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts [hunk 3]
```diff
@@ -18,6 +18,9 @@ import {
   tap,
 } from 'rxjs/operators';
 
+import {
+  FocusConfig,
+} from '@spartacus/storefront';
```
Comment (BenjaminEmiliani) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts]: Make sure to apply our Prettier format to this file (and any other file modified). And typically we don't leave a space in between import lines, instead leave empty between imports and @Component decorator

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts [hunk 4]
```diff
@@ -31,6 +34,13 @@ export class AddedToCartDialogComponent {
   addedEntryWasMerged$: Observable<boolean>;
   promotionLocation: PromotionLocation = PromotionLocation.ActiveCart;
 
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: "button",
+    focusOnEscape: true,
+  };
```
Comment (BenjaminEmiliani) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts]: I have been testing this component with this change, using the focus config, and noticed a change to a desired existing behaviour. If you look in `added-to-cart-dialog.component.html` line 83, the `<a>` has `autofocus` property so that each time the modal is opened the focus shifts to that element. Now using this config the focus is lost after closing with esc key. 

The way to see what I mean is click on 'Add to Cart' for a product. Then hit esc key, and click 'Add to Cart' again. Notice that the 'View Cart' is not focused on second pop up (also esc key is not handled on 2nd attempt)

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.html [hunk 5]
```diff
@@ -2,6 +2,7 @@
   [cxFocus]="focusConfig"
   (esc)="close('Escape clicked')"
   class="cx-import-entries-dialog"
+
```
Comment (BenjaminEmiliani) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.html]: I think this line insertion can be deleted right?

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 6]
```diff
@@ -43,7 +61,13 @@ export class ImportEntriesDialogComponent {
   context$: Observable<AddOrderEntriesContext> =
     this.launchDialogService.data$.pipe(pluck('orderEntriesContext'));
 
-  constructor(protected launchDialogService: LaunchDialogService) {}
+  constructor(
+    protected el: ElementRef,
+    protected launchDialogService: LaunchDialogService
+  ) {}
+  ngOnDestroy(): void {
+    throw new Error('Method not implemented.');
```
Comment (BenjaminEmiliani) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: I don't follow here. What is the purpose of this error throw?

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 1 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: Removed.

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 7]
```diff
@@ -7,28 +14,39 @@ import {
...
 })
-export class ImportEntriesDialogComponent {
+export class ImportEntriesDialogComponent implements OnDestroy{
   iconTypes = ICON_TYPE;
   focusConfig: FocusConfig = {
     trap: true,
     block: true,
-    autofocus: 'button',
-    focusOnEscape: true,
+    autofocus: false,
```
Comment (BenjaminEmiliani) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: I suggest using `autofocus: 'button.btn-action'` instead here. Is there a reason why you changed it to `false`?

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 8]
```diff
@@ -7,28 +14,39 @@ import {
...
+
 import {
   FocusConfig,
   ICON_TYPE,
   LaunchDialogService,
 } from '@spartacus/storefront';
+
 import { BehaviorSubject, Observable } from 'rxjs';
 import { finalize, pluck } from 'rxjs/operators';
-
```
Comment (BenjaminEmiliani) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts:17]: Add this empty line back and remove those in between import statements

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 6 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: I think it was put there when the method was automatically generated. Should have taken it out.

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.html (hunk 5 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.html]: Yes

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 9]
```diff
@@ -43,7 +61,13 @@ export class ImportEntriesDialogComponent {
   context$: Observable<AddOrderEntriesContext> =
     this.launchDialogService.data$.pipe(pluck('orderEntriesContext'));
 
-  constructor(protected launchDialogService: LaunchDialogService) {}
+  constructor(
+    protected el: ElementRef,
```
Comment (BenjaminEmiliani) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: This addition to the constructor requires updating the migration schematics. It's simple to do but requires non trivial work to do, so feel free to ping me on Slack for help. As a tip look at file `projects/schematics/src/migrations/5_0/constructor-deprecations/constructor-deprecations.ts` and look at the array `CONSTRUCTOR_DEPRECATIONS_DATA`. It's where we add components with a deprecated constructor

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts (hunk 3 above)
Comment (echianese) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts]: got it

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html (hunk 2 above)
Comment (echianese) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: 1. Reason changed to: "ESC pressed"
2.  I think ` AddedToCartDialogComponent ` is needed. It was there originally, and without it the dialog is not correctly laid out.

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts (hunk 4 above)
Comment (BenjaminEmiliani) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts]: ^^never mind, even on `develop` the focus shift on 'View Cart' only occurs first time modal is opened.

So using focus config is not a conflict but I don't think we need all for properties that are added.

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 7 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: Thanks. autofocus: 'button.btn-action' is the right thing.  It puts the focus on the "Select File" button which is where I wanted it but I did not know how.  

When I set autofocus to true the focus was placed on the dismiss "+" button which is a poor choice. Autofocus false left the dialog unfocused which is nicer.

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html (hunk 2 above)
Comment (BenjaminEmiliani) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: Hey @echianese I tested something and it seems to work well, can you test it?

Remove these two lines from #dialog
`[cxFocus]="focusConfig"
 (esc)="dismissModal('gone')"`

And and these two lines to `<div class="cx-dialog-buttons"` on line 72
```
(esc)="dismissModal('ESC pressed')"
[cxFocus]="{ autofocus: true }"
```

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts (hunk 4 above)
Comment (BenjaminEmiliani) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.ts]: This comment addresses this: https://github.com/SAP/spartacus/pull/15390/files#r828395795

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html (hunk 2 above)
Comment (echianese) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: Sure, let me make the changes and see. AddToCart dialog works properly now.

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 9 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: I think this is the last comment I need to address. I'll reach out after I look at` constructor-deprecations.ts` .

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 8 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts:17]: done

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 9 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: @BenjaminEmiliani   I took a stab at updating the migration schematics.  Can you take a look?
thx

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html [hunk 10]
```diff
@@ -1,4 +1,4 @@
-<div #dialog>
+<div #dialog class="AddedToCartDialogComponent">
```
Comment (Pucek9) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: Why do you add this class? I don't see usage. 
Additionally, it doesn't comply with our naming convention (`cx-class-name`)

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 9 above)
Comment (Pucek9) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: Usually, new dependencies are added after the previous ones, I'm not sure if it's a requirement, but it's better to check or replace it.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+} from '../../../../shared/constants';
+import { SPARTACUS_CORE } from '../../../../shared/libs-constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const IMPORT_ENTRIES_DIALOG_COMPONENT_MIGRATION: ConstructorDeprecation =
+  {
+    //projects/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts
+    class: IMPORT_ENTRIES_DIALOG_COMPONENT,
+    importPath: SPARTACUS_CORE,
+    deprecatedParams: [],
```
Comment (Pucek9) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]: There should be a list of all params existing previously 
```suggestion
    deprecatedParams: [
      {
        className: LAUNCH_DIALOG_SERVICE,
        importPath: SPARTACUS_STOREFRONTLIB,
      },
    ],
```

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 12]
```diff
@@ -14,21 +20,28 @@ import {
...
+    focusOnEscape: false,
   };
 
+  @HostListener('click', ['$event'])
+  handleClick(event: UIEvent): void {
+    // Close on click outside the dialog window
+    if ((event.target as any).tagName === this.el.nativeElement.tagName) {
+      this.close('Cross click');
+    }
+  }
```
Comment (Pucek9) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: Closing the modal while uploading should be disallowed at all.

**Review by Pucek9 — DISMISSED** [review]
Please follow the comments

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html (hunk 10 above)
Comment (echianese) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: That's a mistake.  Not sure why its there & has been removed.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts (hunk 11 above)
Comment (echianese) [reply projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]: I deprecated LAUNCH_DIALOG_SERVICE and added it back after my new ELEMENT_REF parameter which is first.  Correct?

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 12 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: Will disable close while upload in progress & maybe add a spinner.

Comment (Pucek9) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: ok

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    ],
+    addParams: [
+      {
+        className: ELEMENT_REF,
+        importPath: SPARTACUS_CORE,
+      },
+      {
+        className: LAUNCH_DIALOG_SERVICE,
+        importPath: SPARTACUS_STOREFRONTLIB,
+      },
```
Comment (Pucek9) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]:
```suggestion
```

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+  LAUNCH_DIALOG_SERVICE,
+} from '../../../../shared/constants';
+import { SPARTACUS_CORE, SPARTACUS_STOREFRONTLIB } from '../../../../shared/libs-constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const IMPORT_ENTRIES_DIALOG_COMPONENT_MIGRATION: ConstructorDeprecation =
+  {
+    //projects/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts
+    class: IMPORT_ENTRIES_DIALOG_COMPONENT,
+    importPath: SPARTACUS_CORE,
```
Comment (Pucek9) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]:
```suggestion
    //feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts
    class: IMPORT_ENTRIES_DIALOG_COMPONENT,
    importPath: SPARTACUS_CART_IMPORT_EXPORT_COMPONENTS,
```
and add
```
export const SPARTACUS_CART_IMPORT_EXPORT_COMPONENTS = `${SPARTACUS_CART}/import-export/components`;
```
in `projects/schematics/src/shared/libs-constants.ts`

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 15]
```diff
@@ -43,7 +56,11 @@ export class ImportEntriesDialogComponent {
   context$: Observable<AddOrderEntriesContext> =
     this.launchDialogService.data$.pipe(pluck('orderEntriesContext'));
 
-  constructor(protected launchDialogService: LaunchDialogService) {}
+  constructor(
+    protected el: ElementRef,
+    protected launchDialogService: LaunchDialogService
+  ) {}
```
Comment (Pucek9) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]:
```suggestion
  constructor(
    protected launchDialogService: LaunchDialogService,
    protected el: ElementRef
  ) {}
```

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 16]
```diff
@@ -14,21 +20,28 @@ import {
...
 import { BehaviorSubject, Observable } from 'rxjs';
 import { finalize, pluck } from 'rxjs/operators';
-
 @Component({
   selector: 'cx-import-entries-dialog',
   templateUrl: './import-entries-dialog.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ImportEntriesDialogComponent {
+export class ImportEntriesDialogComponent implements OnDestroy {
```
Comment (Pucek9) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: it seems as unused (`ngOnDestroy(): void {}`) so you can revert this changes

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts (hunk 11 above)
Comment (Pucek9) [reply projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]: You added `LAUNCH_DIALOG_SERVICE` to `addParams` and it's wrong 
Please see suggestions:
https://github.com/SAP/spartacus/pull/15390#discussion_r830300275
https://github.com/SAP/spartacus/pull/15390#discussion_r830300275

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+
+export const IMPORT_ENTRIES_DIALOG_COMPONENT_MIGRATION: ConstructorDeprecation =
+  {
+    //feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts
+    class: IMPORT_ENTRIES_DIALOG_COMPONENT,
+    importPath: SPARTACUS_CART_IMPORT_EXPORT_COMPONENTS,
+    deprecatedParams: [
+      {
+        className: IMPORT_ENTRIES_DIALOG_COMPONENT,
+        importPath: SPARTACUS_CART_IMPORT_EXPORT_COMPONENTS,
```
Comment (Pucek9) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]: I think you have a mistake here. I suggested removing LAUNCH_DIALOG_SERVICE from `addParams`, it should stay here. I recommend github's "commit suggestion" option for avoiding confusion (then manually fix imports etc)
```suggestion
        className: LAUNCH_DIALOG_SERVICE,
        importPath: SPARTACUS_STOREFRONTLIB,
```

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts (hunk 11 above)
Comment (echianese) [reply projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]: Thanks.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts (hunk 17 above)
Comment (echianese) [reply projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]: Right.  The 1'st  parameter is the dialog service, not the component itself. I pushed up the correction.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts (hunk 13 above)
Comment (echianese) [reply projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]: Done

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 15 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: done

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 16 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: done

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 12 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: done

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts (hunk 14 above)
Comment (echianese) [reply projects/schematics/src/migrations/5_0/constructor-deprecations/data/import-entries-dialog.component.migration.ts]: done

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 18]
```diff
@@ -25,10 +29,21 @@ export class ImportEntriesDialogComponent {
...
+    autofocus: 'button.btn-action',
+    focusOnEscape: false,
   };
 
+  uploading: boolean = false;
+  @HostListener('click', ['$event'])
+  handleClick(event: UIEvent): void {
+    // Close on click outside the dialog window
+    if ((event.target as any).tagName === this.el.nativeElement.tagName) {
+      if (!this.uploading) {
```
Comment (Pucek9) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]:
```suggestion
      if (!this.summary$.getValue().loading) {
```
and please remove `uploading` property

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html [hunk 19]
```diff
@@ -76,7 +76,8 @@
               cxModal="dismiss"
               cxModalReason="View Cart click"
               class="btn btn-primary"
-              autofocus
+              (esc)="dismissModal('ESC pressed')"
```
Comment (Pucek9) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html:79]: > Opened modals should allow to close them on ESC key press.

1 I can not close import dialog by escape after upload (when seeing summary) - need to click the mouse on modal first or press tab first (but it will work only if `show more` button exists)

I think trigger focus to the element is necessary when finished.

2 I can still use escape while uploading when click on modal first (or press tab if `show more` button exists)

something like this could help for point 2:
image

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts [hunk 20]
```diff
@@ -14,7 +19,6 @@ import {
 } from '@spartacus/storefront';
 import { BehaviorSubject, Observable } from 'rxjs';
 import { finalize, pluck } from 'rxjs/operators';
-
 @Component({
   selector: 'cx-import-entries-dialog',
   templateUrl: './import-entries-dialog.component.html',
```
Comment (Pucek9) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts:24]: > Opened modals should allow to close them on ESC key press.

1 I can not close import dialog by escape after upload (when seeing summary) - need to click the mouse on modal first or press tab first (but it will work only if `show more` button exists)

I think trigger focus to the element is necessary when finished.

2 I can still use escape while uploading when click on modal first (or press tab if `show more` button exists)

something like this could help for point 2:
image

Comment (Pucek9) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts:25]: > Opened modals should allow to close them on ESC key press.

1 I can not close import dialog by escape after upload (when seeing summary) - need to click the mouse on modal first or press tab first (but it will work only if `show more` button exists)

I think trigger focus to the element is necessary when finished.

2 I can still use escape while uploading when click on modal first (or press tab if `show more` button exists)

something like this could help for point 2:
image

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html (hunk 10 above)
Comment (Pucek9) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: Yeah, I also don't know why it is still here 😅

File: feature-libs/cart/import-export/styles/_import-entries-dialog.scss [hunk 21]
```diff
@@ -1,14 +1,11 @@
 cx-import-entries-dialog {
   background-color: rgba(0, 0, 0, 0.5);
-
   .cx-import-entries-dialog {
     @extend .modal-dialog;
     @extend .modal-dialog-centered;
 
     .cx-import-entries-container {
       @extend .modal-content;
-      pointer-events: none;
```
Comment (Pucek9) [inline feature-libs/cart/import-export/styles/_import-entries-dialog.scss:10]: I can see that the ability to turn off the modal by clicking the mouse outside of it is associated with the removal of another functionality (equally important, maybe even more important) - selecting and copying text/output.

In this case, we especially need consent to the changes @Xymmer

Comment (Pucek9) [reply feature-libs/cart/import-export/styles/_import-entries-dialog.scss:10]: Update: I can see in add to cart dialog, we are able to both: select text and close by clicking outside. Can't you do the same here? 

Oh, it's even possible in summary (but not immediately after opening dialog)

Comment (echianese) [reply feature-libs/cart/import-export/styles/_import-entries-dialog.scss:10]: Adding  "tabindex="0" to the cx-import-entries-container lets the user select text

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 20 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts:25]: I fixed point 1, and used your suggestions for point 2 which prevents the dialog from closing while uploading.

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html (hunk 10 above)
Comment (echianese) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: Gone now for real.

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts (hunk 9 above)
Comment (echianese) [reply feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts]: I fixed the parameter list so it's in the preferred order.

File: feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.html [hunk 22]
```diff
@@ -43,9 +45,11 @@
 
       <ng-template #importSummary>
         <cx-import-entries-summary
-          [summary]="summary$ | async"
+          [cxFocus]="focusConfig"
+          [summary]="summary"
           [type]="context.type"
           (closeEvent)="close('Close Import Products Dialog')"
+
```
Comment (Pucek9) [inline feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.html]:
```suggestion
```

File: projects/storefrontlib/layout/a11y/keyboard-focus/autofocus/auto-focus.directive.ts [hunk 23]
```diff
@@ -78,7 +84,10 @@ export class AutoFocusDirective
    * focusable child element will be focussed.
    */
   handleFocus(event?: KeyboardEvent) {
-    if (this.shouldAutofocus) {
+    // Dalvir
+    console.log(event?.key)
```
Comment (Pucek9) [inline projects/storefrontlib/layout/a11y/keyboard-focus/autofocus/auto-focus.directive.ts]:
```suggestion
```

Comment (Dalvir-Personal) [reply projects/storefrontlib/layout/a11y/keyboard-focus/autofocus/auto-focus.directive.ts]: Sorry about this. Fixed

**Comment by Pucek9** [comment]
I can se below issues:
- **add to cart dialog**: closing by `esc` keydown works only if focused on `view cart` button, pressing esc on other elements causes losing focus on dialog but it's not closed
- **import products dialog**: when see summary, I can not navigate to `X` button by keyboarding (trap between `show` links and `close` button)
- **AnonymousConsentDialogComponent** I don't see visual focus
- **replanishment** dialog has no `x` button
